docs/2.10.0/docs/canton/usermanual/contract_keys.rst

```diff
@@ -278,8 +278,8 @@ Caveats to keep in mind are:
   However, this is easy to change.
 - This approach relies on a particular internal behavior of Canton (as discussed below).
   While we don't expect the behavior to change, we do not currently make strong guarantees that it will not change.
-- If the participant is connected to multiple sync domains, the approach may fail in future versions of Canton.
-  To be future-proof, you should only use it in the settings when your participant is connected to a single sync domain.
+- If the participant is connected to multiple sync domains, the approach will fail in future versions of Canton.
```

Comment:
how about a short description why it will fail, something like, "because the Generator contract may be reassigned to another sync domain without the Keyed contract being reassigned, which will return None by lookupByKey".

Reply:
Not reassigning the `Keyed` contract is not the problem here. This will break because of out-of-order publishing of events across domains. I'll add a sentence.